### ISSUE TITLE
Discourage order editing in order screen when order is disputed

### ISSUE THREAD

**Issue by haszari** [Automattic/woocommerce-payments#7567]
### Describe the bug

Follow up issue from 

- #6378

There are various elements/features in the order screen that we could disable (lock) when the order is disputed. 

The goal of this issue is to implement this locking as appropriate, and give the merchant info (e.g. via tooltips) so they understand why editing is locked. Note that it might not be practical to disable everything, and also there may be some limitations with where we can show tooltips. 

More info from @nikkivias :

There will be many places in the order page that will be disabled during disputes. So we added this message in the banner to proactively communicate about the page's disabled state rather than having the user poke around and find out. 

In line with this move, RiEQmaKRI7u9USAI3lyZz0VX-fi-7596%3A86883 recommended removing the pencil icons from places like Billing and Shipping. Let's forgo that so we can trigger 'not allowed' tooltips instead.

![Screenshot 2023-10-27 at 1 02 59 AM](https://github.com/Automattic/woocommerce-payments/assets/99638528/53682683-1ef0-472f-adbc-1e81ebd93e37)
![Screenshot 2023-10-27 at 1 05 39 AM](https://github.com/Automattic/woocommerce-payments/assets/99638528/196db9e7-8d16-437d-a040-44244b87071f)

We will want to implement tooltips anywhere that is disabled in the interface, but instead of consolidating the message, let's make it specific to the UI elements. For example, if they hover over the Refund button, we say 'Refunding is disabled during disputes' and if they try an editing elements, we say 'Order editing is disabled during disputes'.

---

#### more context
This is a follow in from a decision to disable the refund button specifically. The bigger picture here is that we are patching the WooCommerce core edit screen because we (WooPayments) know more about the order – i.e. that it is disputed. 

Because we're patching this piecemeal, the UI will likely feel somewhat disconnected, and we might not actually be able to **stop** merchants from taking actions (e.g refunding or editing order via other mechanisms). 

Ultimately, the best way to fix this problem is in WooCommerce core – e.g. an order status. We'll need to keep this in mind if/when we work on this… i.e. 

- If we can do some lean elegant fixes to discourage editing, great.
- If we have to do complex hacks to disable or show tooltips, the benefit might not be worth the cost.

Here's an example of a UX tradeoff in the current implentation: the `Refund` button is disabled only after the dispute notice has loaded and displayed. Not ideal.

### To Reproduce
- View `WooCommerce > Orders > order detail` for a order with an active dispute. 
- If the dispute is not resolved, the merchant generally should not refund or edit the order as the transaction is now out of their control.

### Actual behavior
- Can edit some order data
- Refund button is disabled (when notice is displayed)

**Comment by haszari** [comment]
I don't think this is a high priority. Essentially all these changes are a band-aid to cover the fact that WooCommerce core doesn't understand that the order is disputed. The long term fix would be a new order status; I don't think we should invest significant time in disabling specific UI elements and showing tooltips.

**Comment by haszari** [comment]
@souravdebnath1986  I'm thinking we can close this. We've done a first pass of the order screen to ensure that merchant knows there is a live dispute, and shouldn't refund it. 

If we were to invest more time here I think we should tackle this from the core end, with an appropriate order status, rather than band-aid patch the UI from WooPayments plugin. That would be a project, i.e. an idea for the roadmap. 

With that in mind, I don't think we should prioritise anything from this issue for maintenance. Closing – feel free to reopen or open a new issue if there are robust improvements we could make to discourage refunding disputed orders. Also if this becomes a significant pain point (lots of merchants are getting in trouble by refunding disputed orders) then we could take another look at the priority of this.
